cmd/kubeadm/app/cmd/phases/reset/unmount_linux.go

```diff
@@ -45,8 +45,10 @@ func unmountKubeletDirectory(absoluteKubeletRunDirectory string) error {
 		if len(m) < 2 || !strings.HasPrefix(m[1], absoluteKubeletRunDirectory) {
 			continue
 		}
-		if err := syscall.Unmount(m[1], 0); err != nil {
+		if err := syscall.Unmount(m[1], syscall.MNT_DETACH); err != nil {
 			klog.Warningf("[reset] Failed to unmount mounted directory in %s: %s", absoluteKubeletRunDirectory, m[1])
+			// return error to terminate reset action to avoid recursive deleting contents within any active mount points.
+			return err
```

Comment:
returning an error is the right thing to do, but this is a behavior change.

"kubeadm reset" has a force flag -f and perhaps that can mean using MNT_FORCE.

or instead of returning the error, we need to rework the logic to leave mounts that failed but clean everything else in the run dir

Reply:
gotcha, but It would take a lot of work to clean up paths other than the mountpoint

Reply:
syscall.Unmount(m[1], syscall.MNT_DETACH) most likely detach the mount point from filesystem to prevent recursive deleting even files/dir related to mount point is opened by  running process or terminal. If we could detach the mount point, it is safe way to keep /var/lib/kubelet data rather than make further data damage!

Reply:
> gotcha, but It would take a lot of work to clean up paths other than the mountpoint

i am not saying this is what we have to do but, i don't think it's that complicated.

- list all mount paths
- find which we cannot unmount, show warnings for them, don't error
- clean all dir / files except mnt paths
- leave the mount dir without cleaning

TMK, the mnt dir will only have the problem mounts remaining. the rest of the mnt paths will be removed on unmount.

Reply:
yeah, I think this is a more thorough approach. but I understand that our priority at this point is to stop the bleeding, so can this pr be merged first?

Reply:
We'll put up another pr to do what you're talking about. @neolit123

Reply:
> yeah, I think this is a more thorough approach. but I understand that our priority at this point is to stop the bleeding, so can this pr be merged first?

my vote would be -1. i don't like this approach too much. you could patch your local build of kubeadm to stop the bleeding.
this particular fix will be available in 1.29.

we haven't decided:
- what exactly to do as a fix
- whether we are going to backport this fix to < 1.29

Reply:
> whether we are going to backport this fix to < 1.29

Yes, we're experiencing this problem too, it affects too many versions to do a fix for each one. We've wrapped a shell around kubeadm to stop the bleeding.

Reply:
> > yeah, I think this is a more thorough approach. but I understand that our priority at this point is to stop the bleeding, so can this pr be merged first?
> 
> my vote would be -1. i don't like this approach too much. you could patch your local build of kubeadm to stop the bleeding. this particular fix will be available in 1.29.

okay, Since the community isn't in a hurry to stop the bleeding, we'll modify the pr to meet your needs.